cfme/common/provider.py

```diff
@@ -277,7 +298,11 @@ def _do_stats_match(self, client, stats_to_match=None, refresh_timer=None, db=Tr
 
         for stat in stats_to_match:
             try:
-                cfme_stat = getattr(self, stat)(db=db)
+                if method:
```

Comment:
jkrocil: so this is what you get when you run validate
jkrocil: `ProviderHasNoProperty: Provider does not know how to get 'num_host'`
jkrocil: but this is what is actually happening
jkrocil: `AttributeError: No such attribute/subcollection ems_id`
jkrocil: hope this helps

The exception is being incorrectly interpreted by the `except AttributeError:` block below.

Tested against vsphere55, cfme 5.5.0.10.